feathr-impl/src/test/scala/com/linkedin/feathr/offline/AnchoredFeaturesIntegTest.scala

```diff
@@ -386,6 +386,146 @@ class AnchoredFeaturesIntegTest extends FeathrIntegTest {
+          |         default: -1
+          |      }
+          |      featureWithNull3: {
+          |         def: "isPresent(value) ? toNumeric(value) : 0"
+          |         type: CATEGORICAL
+          |         default: "null"
+          |      }
+          |      featureWithNull4: {
+          |         def: "isPresent(value) ? toNumeric(value) : 0"
+          |         type: TERM_VECTOR
+          |         default: {}
+          |      }
+          |      featureWithNull6: {
+          |         def: "isPresent(value) ? toNumeric(value) : 0"
+          |         type: DENSE_VECTOR
+          |         default: [1, 2, 3]
+          |      }
+          |      featureWithNull5: {
+          |         def: "isPresent(value) ? toNumeric(value) : 0"
+          |         default: 1
+          |      }
+          |    }
+          |  }
+          |
+          |  anchor2: {
+          |    source: "anchorAndDerivations/nullValueSource.avro.json"
+          |    key: "toUpperCaseExt(mId)"
+          |    features: {
+          |      featureWithNull2: "isPresent(value) ? toNumeric(value) : 0"
+          |    }
+          |  }
+          |  swaAnchor: {
+          |    source: "swaSource"
+          |    key: "x"
+          |    features: {
+          |      aEmbedding: {
+          |        def: "embedding"
+          |        aggregation: LATEST
+          |        window: 3d
+          |        default: 2
+          |      }
+          |    }
+          |  }
+          |  swaAnchor1: {
+          |    source: "swaSource1"
+          |    key: "x"
+          |    features: {
+          |      memberEmbeddingAutoTZ: {
+          |        def: "embedding"
+          |        aggregation: LATEST
+          |        window: 3d
+          |        type: {
+          |          type: TENSOR
+          |          tensorCategory: SPARSE
+          |          dimensionType: [INT]
+          |          valType: FLOAT
+          |        }
+          |      }
+          |    }
+          |  }
+          |}
+          |derivations: {
+          |
+          | derived_featureWithNull: "featureWithNull * 2"
+          | derived_featureWithNull2: "featureWithNull2 * 2"
+          |}
+        """.stripMargin,
+      observationDataPath = "anchorAndDerivations/testMVELLoopExpFeature-observations.csv")
+
+    df.data.show()
+    val featureList = df.data.collect().sortBy(row => if (row.get(0) != null) row.getAs[String]("a_id") else "null")
+    assertEquals(featureList(0).getAs[Row]("aEmbedding"),
+      Row(mutable.WrappedArray.make(Array("")), mutable.WrappedArray.make(Array(2.0f))))
+    assertEquals(featureList(0).getAs[Row]("featureWithNull3"), "null")
+    assertEquals(featureList(0).getAs[Row]("featureWithNull5"), mutable.Map("" -> 1.0f))
+    assertEquals(featureList(0).getAs[Row]("featureWithNull"),-1.0f)
+    assertEquals(featureList(0).getAs[Row]("featureWithNull4"),Map())
+    assertEquals(featureList(0).getAs[Row]("featureWithNull2"),1.0f)
```

Comment:
What happens when we try to derive a feature from a feature that has default values such as a  null or empty Map? Are those already handled?

Reply:
Added a test case for that.